prov-api/dependencies/dependencies.py

```diff
@@ -36,3 +42,79 @@
 
 def user_is_admin(user: User) -> bool:
     return admin_usage_role in user.roles
+
+
+def build_kms_service_from_config(config: Config) -> KMSEncryptionService:
+    """
+
+    Builds the AWS Key Management service encryption service from the config object.
+
+    Parameters
+    ----------
+    config : Config
+        Config to build from
+
+    Returns
+    -------
+    KMSEncryptionService
+        The encryption service
+    """
+    return KMSEncryptionService(KMSConfig(
+        key_id=config.user_key_id,
+        region=config.user_key_region
+    ))
+
+
+def get_encryption_service(config: Config = Depends(get_settings)) -> EncryptionService:
+    """
+    FastAPI dependency injection to get encryption service. This exposes
+    abstract base class implementation so we can swap out to other encryption
+    service implementations as needed.
+
+    Parameters
+    ----------
+    config : Config, optional
+        The config - this is injected prior to the encryption service, by default Depends(get_settings)
+
+    Returns
+    -------
+    EncryptionService
+        The resulting service as an EncryptionService type
+    """
+    return build_kms_service_from_config(config)
+
+
+def get_user_context_header(user_cipher: str, config: Config) -> Dict[str, str]:
+    """
+
+    Returns a dictionary intended to be merged with other headers which includes
+    the user cipher at the configured user context header field.
+
+    Parameters
+    ----------
+    user_cipher : str
+        The encrypted user context
+    config : Config
+        The config
+
+    Returns
+    -------
+    Dict[str, str]
+        The headers
+    """
+    return {config.user_context_header: user_cipher}
+
+
+async def get_user_cipher(
```

Comment:
noticed that some of these functions appear in multiple service dependencies the same. Is it possible to move them into the shared functinoality?

Reply:
I moved as much as I could to shared, to do more I'd need to know about the API's config objects as part of the shared interface in order to get the configured header, and other stuff.